src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.WorkerThread.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using Internal.Runtime.Augments;
+
+namespace System.Threading
+{
+    internal static partial class ClrThreadPool
+    {
+        internal static class WorkerThread
+        {
+            internal static WaitSubsystem.WaitableObject s_semaphore = WaitSubsystem.WaitableObject.NewSemaphore(0, int.MaxValue);
+
+            private const int TimeoutMs = 20 * 1000;
+            
+            private static void WorkerThreadStart()
+            {
+                RuntimeThread.InitializeThreadPoolThread();
+                // TODO: Event: Worker Thread Start event
+
+                while (true)
+                {
+                    // TODO: Event:  Worker thread wait event
+                    while (WaitSubsystem.Wait(s_semaphore, TimeoutMs, false, true))
```

Comment:
This would reduce the semaphore's signal count indicating that the thread is now active. Inside this loop there should be another loop that processes work until Dispatch returns false.

Also, the working thread count should be incremented before this inner loop to actually indicate that there is one more working thread.